ctc_lib/container_tool_command.go

```diff
@@ -0,0 +1,110 @@
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctc_lib
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/help"
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/version"
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/writer"
+	"github.com/spf13/cobra"
+)
+
+type ContainerToolCommand struct {
+	*cobra.Command
+	Phase                string
+	Version              string
+	HelpText             string
+	Output               interface{}
+	subCommandsOutputMap map[string]interface{}
+	args                 []string
+}
+
+type ContainerToolCommandList struct {
+	*cobra.Command
+	Phase                string
+	Version              string
+	OutputList           []interface{}
+	subCommandsOutputMap map[string]interface{}
+	args                 []string
+}
+
+// Initializes the Root Command with Common subcommands and Configs.
+func (ctc *ContainerToolCommand) init() {
+	// Add version subcommand
+	ctc.AddCommand(version.NewVersionCommand(ctc.Version,
+		ctc.Command.Name()).Command, &version.VersionOutput{})
+
+	// Add help subcommand
+	ctc.Command.SetHelpCommand(help.NewHelpCommand(ctc.HelpText,
+		ctc.Command.Name()).Command)
+
+	// Set up Root Command Help
+	ctc.Command.SetHelpTemplate(help.HelpTemplate)
+}
+
+func (ctc *ContainerToolCommand) AddCommand(command *cobra.Command, output interface{}) {
+	ctc.Command.AddCommand(command)
+	ctc.SetSubCommandsOutputMap(command.Name(), output)
+}
+
+// Set Args for Command. Useful for testing.
+func (ctc *ContainerToolCommand) SetArgs(args []string) {
+	ctc.args = args
+	ctc.Command.SetArgs(args)
+}
+
+func (ctc *ContainerToolCommand) SetSubCommandsOutputMap(name string, output interface{}) {
+	if ctc.subCommandsOutputMap == nil {
+		ctc.subCommandsOutputMap = make(map[string]interface{})
+	}
+	ctc.subCommandsOutputMap[name] = output
+}
+
+func (ctc *ContainerToolCommand) GetSubCommandsOutputMap() map[string]interface{} {
+	return ctc.subCommandsOutputMap
+}
+
+func (ctc *ContainerToolCommand) Execute() {
+	ctc.init()
+	if err := ctc.Command.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+// Executes the command and returns a object.
+func (ctc *ContainerToolCommand) ExecuteO() interface{} {
+	ctc.init()
+	// Initilize Object Writer.
+	ctcWriter := writer.NewCTCBuffer(ctc.Output)
+	ctc.Command.SetOutput(ctcWriter)
+	err := ctc.Command.Execute()
+	if err != nil {
+		return err
+	}
+
+	targetCommand, _, _ := ctc.Command.Find(ctc.args)
+	output := ctc.GetSubCommandsOutputMap()[targetCommand.Name()]
+	err = ctcWriter.ReadObject(output)
```

Comment:
I think it would be more idiomatic go to reuse the standard buffer types, and implement the parsing here in a separate function.